### PR TITLE
Move case attachments to blob db

### PR DESCRIPTION
**Pull request by millerdev**
Probably best viewed by commit 🐠 

@snopoke @dannyroberts

### CODE REVIEW

File: corehq/blobs/migrate.py
Comment (millerdev) [inline corehq/blobs/migrate.py]: I'm pretty sure these are the only classes that need to be migrated. There is a [`SupplyPointCase`](https://github.com/dimagi/commcare-hq/blob/c334bd9ed7f1c8dd7a8ac8329237349a68d4b3a2/corehq/apps/commtrack/models.py#L285) subclass of `CommCareCase`, but it looks like it's never saved with that doctype (at least there were none saved in couch on prod when I looked last week). Also there are a bunch of subclasses in the `custom` code dir, but it looked like they were all reporting helpers and never saved (and there were none saved in prod couch).

File: corehq/apps/app_manager/models.py
Comment (millerdev) [inline corehq/apps/app_manager/models.py]: Previously this had always evaluated to `True` because jsonobject replaced the class attribute boolean value with a `jsonobject.properties.BooleanProperty`. It now works as expected, which is why `_attachments` had to be removed from some test JSON docs.

File: corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py
Comment (millerdev) [inline corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py]: This test was failing for me because I have a service running on `localhost:9000`, and the ES service was running in boot2docker, which has a different IP address than localhost.

File: corehq/blobs/mixin.py
Comment (snopoke) [inline corehq/blobs/mixin.py]: Why does this have to happen outside of `atomic_blobs` context?

File: corehq/apps/app_manager/models.py
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: I've been bitten by this before.

**Comment by snopoke** [comment]
:boot: :building_construction:

**Review by snopoke — APPROVED** [review]

File: corehq/blobs/mixin.py
Comment (millerdev) [inline corehq/blobs/mixin.py]: The code in the `atomic_blobs` context is executed [just before](https://github.com/dimagi/commcare-hq/blob/c334bd9ed7f1c8dd7a8ac8329237349a68d4b3a2/corehq/blobs/mixin.py#L206) the [doc is saved](https://github.com/dimagi/commcare-hq/blob/c334bd9ed7f1c8dd7a8ac8329237349a68d4b3a2/corehq/blobs/mixin.py#L207). This is outside the context so they are deleted after successful save. If we deleted them inside the context then they would be deleted even if the save failed.

It is possible that an exception could be raised just outside the context (after the doc is saved) before all deletes have gone through, which would result in orphaned blobs. I think that's not very likely, and better to err on the side of keeping data we don't need than deleting data we might need.

**Comment by millerdev** [comment]
One other thing I forgot to ask about this: With the form attachments migration there were ES indexes to update, but it looks like case attachments are not indexed in ES. Does that sound right, or should this PR be updating ES indexes too?

**Comment by snopoke** [comment]
The only case index that has attachments indexed is `reportcase` but I don't think it needs any changes for this: https://github.com/dimagi/commcare-hq/blob/624a9fc206d39d79761f8ab1cd2cc05e32733a4e/corehq/pillows/mappings/reportcase_mapping.py#L66-L66

**Comment by snopoke** [comment]
This is GTG from me, holding off on merge in case you want to wait for @dannyroberts to do a pass.

**Comment by millerdev** [comment]
Thanks @snopoke. I'm going to wait on @dannyroberts review before merging.

File: corehq/apps/app_manager/models.py
Comment (dannyroberts) [inline corehq/apps/app_manager/models.py]: Very true. When I designed `jsonobject` I was both blessed with having couchdbkit's interface already designed for me to match exactly, and cursed with the blemishes of that design. woh, woh

File: corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py
Comment (dannyroberts) [inline corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py]: I'm not sure I get the `- 2` part

**Review by dannyroberts — APPROVED** [review]

File: corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py
Comment (millerdev) [inline corehq/ex-submodules/pillowtop/tests/test_elasticsearch.py]: Random different port than the configured one. This is not a great feature of the way this test is written because for any other random port there could be another service running that would cause an incorrect failure.

**Pull request merged**
